krystex/src/test/java/com/flipkart/krystal/krystex/node/KrystalNodeExecutorTest.java

```diff
@@ -50,6 +51,47 @@ void tearDown() {
     this.krystalNodeExecutor.close();
   }
 
+  /** Executing same node multiple times in a single execution */
+  @Test
+  void multiRequestExecution() throws Exception {
+    NodeDefinition nodeDefinition =
+        nodeDefinitionRegistry.newNodeDefinition(
+            "node",
+            newComputeLogic(
+                    "nodeLogic", Collections.emptySet(), dependencyValues -> "computed_value")
+                .nodeLogicId());
+
+    CompletableFuture<Object> future_1 =
+        krystalNodeExecutor.executeNode(nodeDefinition.nodeId(), Inputs.empty(), "req_1");
+    CompletableFuture<Object> future_2 =
+        krystalNodeExecutor.executeNode(nodeDefinition.nodeId(), Inputs.empty(), "req_2");
+
+    krystalNodeExecutor.flush();
+    assertEquals("computed_value", timedGet(future_1));
+    assertEquals("computed_value", timedGet(future_2));
+  }
+
+  @Test
+  void multiRequestExecutionWithNullRequestId() throws Exception {
+    NodeDefinition nodeDefinition =
+        nodeDefinitionRegistry.newNodeDefinition(
+            "node",
+            newComputeLogic(
+                    "nodeLogic", Collections.emptySet(), dependencyValues -> "computed_value")
+                .nodeLogicId());
+
+    CompletableFuture<Object> future_1 =
+        krystalNodeExecutor.executeNode(nodeDefinition.nodeId(), Inputs.empty(), "req_1");
+    try {
+      CompletableFuture<Object> future_2 =
+          krystalNodeExecutor.executeNode(nodeDefinition.nodeId(), Inputs.empty(), null);
+    } catch (Exception e) {
```

Comment:
Catching exceptions in testcases is not recommended.
Use assertThat().throws().containError syntax instead